src/cryptography/hazmat/primitives/asymmetric/padding.py

```diff
@@ -55,14 +61,17 @@ def __init__(self, mgf, algorithm, label):
 class MGF1(object):
     MAX_LENGTH = object()
 
-    def __init__(self, algorithm):
+    def __init__(self, algorithm: hashes.HashAlgorithm):
         if not isinstance(algorithm, hashes.HashAlgorithm):
             raise TypeError("Expected instance of hashes.HashAlgorithm.")
 
         self._algorithm = algorithm
 
 
-def calculate_max_pss_salt_length(key, hash_algorithm):
+def calculate_max_pss_salt_length(
+    key: typing.Union["rsa.RSAPrivateKey", "rsa.RSAPublicKey"],
```

Comment:
Shouldn't need quotes here

Reply:
You, in fact, do because of recursive import stuff -- `padding.py` is imported by `rsa.py` so the module isn't fully initialized.